src/AbstractMetricsManager.h

```diff
@@ -65,7 +65,7 @@ class AbstractMetricsBucket
     const std::bitset<GROUP_SIZE> *_groups;
 
     // merge the metrics of the specialized metric bucket
-    virtual void specialized_merge(const AbstractMetricsBucket &other) = 0;
+    virtual void specialized_merge(const AbstractMetricsBucket &other, bool aggregate) = 0;
```

Comment:
Rather than `bool aggregate`, let's add a new enum class with DEFAULT and SUM items, and have the parameter be `agg_operator` defaulting to DEFAULT. for the quantile merge we will pass SUM